### ISSUE TITLE
TCK's GlobalTagsTest failing with smallrye-config 1.3.11

### ISSUE THREAD

**Issue by jmartisk** [smallrye/smallrye-metrics#205]
```
java.lang.IllegalStateException: Configuration already registered for the given class loader
	at org.eclipse.microprofile.metrics.tck.tags.GlobalTagsTest.customConfigSource(GlobalTagsTest.java:108)
```

**Comment by kenfinnigan** [comment]
Does this require another SmallRye Config fix?

**Comment by jmartisk** [comment]
Looks like this is a problem in the Metrics TCK, this behavior seems compliant with Config spec, at least with what is written in javadoc https://github.com/eclipse/microprofile-config/blob/1.3/api/src/main/java/org/eclipse/microprofile/config/spi/ConfigProviderResolver.java#L82

**Comment by kenfinnigan** [comment]
Thoughts @jmesnil @dmlloyd ?

**Comment by jmartisk** [comment]
Metrics TCK overrides the existing config but that throws an exception now, I suppose we will need to challenge the TCK to call `release` on the existing config, then register the new one, then release the new one and re-register the old one. Can't say for sure how Quarkus will behave in that case though

**Comment by dmlloyd** [comment]
The entire register/release edifice in MP config is flawed (as you've discovered), but it's unlikely that we'll ever get the spec cleaned up in this regard.  Releasing the existing config is the correct behavior IMO, so I'd go ahead and challenge the TCK.  I'd recommend that any test which creates a config should release the config after the test is complete, rather than get+release+register at the top of every test.

**Comment by jmartisk** [comment]
This will be resolved by upgrading to MP Metrics 2.2.1, which should be out very soon, maybe even today.

**Comment by jmartisk** [comment]
For master branch it will be fixed by https://github.com/smallrye/smallrye-metrics/pull/219
